sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageProcessor.java

```diff
@@ -0,0 +1,75 @@
+
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.models.ServiceBusReceiveMode;
+import com.azure.messaging.servicebus.stress.util.EntityType;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+
+/**
+ * Test ServiceBusProcessorClient
+ */
+@Service("MessageProcessor")
+public class MessageProcessor extends ServiceBusScenario {
+    private static final ClientLogger LOGGER = new ClientLogger(MessageProcessor.class);
+
+    @Value("${MAX_CONCURRENT_CALLS:20}")
+    private int maxConcurrentCalls;
+
+    @Value("${PREFETCH_COUNT:0}")
+    private int prefetchCount;
+
+    @Override
+    public void run() {
+        final String connectionString = options.getServicebusConnectionString();
+        final EntityType entityType = options.getServicebusEntityType();
+        String queueName = null;
+        String topicName = null;
+        String subscriptionName = null;
+        if (entityType == EntityType.QUEUE) {
+            queueName = options.getServicebusQueueName();
+        } else if (entityType == EntityType.TOPIC) {
+            topicName = options.getServicebusTopicName();
+            subscriptionName = options.getServicebusSubscriptionName();
+        }
+
+        final String receiveCounterKey = "Number of received messages - "
+            + (queueName != null ? queueName : topicName + "/" + subscriptionName);
+
+        ServiceBusProcessorClient client = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .retryOptions(new AmqpRetryOptions().setTryTimeout(Duration.ofSeconds(5)))
+            .processor()
+            .queueName(queueName)
+            .topicName(topicName)
+            .subscriptionName(subscriptionName)
+            .maxConcurrentCalls(maxConcurrentCalls)
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .disableAutoComplete()
+            .prefetchCount(prefetchCount)
+            .processMessage(messageContext -> {
+                LOGGER.verbose("Before complete. messageId: {}, lockToken: {}",
+                    messageContext.getMessage().getMessageId(),
+                    messageContext.getMessage().getLockToken());
+                messageContext.complete();
+                rateMeter.add(receiveCounterKey, 1);
+                LOGGER.verbose("After complete. messageId: {}, lockToken: {}",
+                    messageContext.getMessage().getMessageId(),
+                    messageContext.getMessage().getLockToken());
+            })
+            .processError(err -> {
+                throw LOGGER.logExceptionAsError(new RuntimeException(err.getException()));
+            })
+            .buildProcessorClient();
+
+        client.start();
```

Comment:
Same with other clients. Does clean-up matter?

Reply:
I'll add `close()` for `sender` and `sync receiver` since we know the total send/receive messages. But for `processor` and `async receiver`, we don't know when we can close it. We keep the processor running until we manually shut down or deploy another test. If someone needs to, they can add a fix duration to `processor` test.  

The clean-up will not impact test running and its results. Because we will run each test scenraio on different resource groups and all resources are auto-deleted after we end the test.